packages/fxa-auth-db-mysql/db-server/test/backend/db_tests.js

```diff
@@ -2597,6 +2597,84 @@ module.exports = function (config, DB) {
       })
     })
 
+    describe('account subscriptions', () => {
+      const subs = [...Array(10)].map(() => 'sub-' + Math.random())
+      let account
+
+      beforeEach(() => {
+        account = createAccount()
+        return db.createAccount(account.uid, account)
+      })
+
+      it('should create a new subscription for account', () => {
+        return db.createAccountSubscription(account.uid, subs[0], 'prod0', Date.now())
+          .then(result => assert.deepEqual(result, {}))
+      })
+
+      it('should fail to create for unknown account', () => {
+        return db.createAccountSubscription('12312312312', subs[1], 'prod2', Date.now())
+          .then(assert.fail, err => assert.equal(err.errno, 116, 'not found'))
+      })
+
+      it('should fail to create duplicate subscription id', () => {
+        const account2 = createAccount()
+        return db.createAccount(account2.uid, account2)
+          .then(() => db.createAccountSubscription(account.uid, subs[2], 'prod2', Date.now()))
+          .then(() => db.createAccountSubscription(account2.uid, subs[2], 'prod3', Date.now()))
```

Comment:
I'm confused by the terminology I think, a subscription is the ID that links a product to an account? When first reading I thought `subscription` was the product. Does the subscription ID come from the payment provider?

Reply:
A subscription is a record that ties together product, plan, and account. Different plans might offer different products at different price points, payment schedules, etc.

The subscription ID and product name will come from subhub. It may or may not pass them along from the payment provider - we should treat them as opaque identifiers. 

We're storing the subscription ID in order to update/cancel the subscription later. We're storing the product name to figure out what capabilities to expose to clients.